api/command/mixins/team.py

```diff
@@ -0,0 +1,503 @@
+                 permissions to create a team, or if the specified user with
+                 lead_id does not have the permission to be a lead
+        :raises: SlackAPIError if a channel name is provided by an error
+                 is encountered retrieving the members of that channel
+        :raises: GithubAPIException if an error occurs on team creation or
+                 Github team member addition
+        :return: True if the team creation was successful, False otherwise
+        """
+        logging.info("Team create command API called")
+        command_user = self._db_facade.retrieve(User, caller_id)
+        logging.debug(f"Calling user: {command_user.__str__()}")
+
+        if not check_permissions(command_user, None):
+            msg = f"Calling user with Slack ID {caller_id} has permission" \
+                f" level {str(command_user.permissions_level)}, " \
+                "insufficient for creating a team!"
+            logging.error(msg)
+            raise PermissionError(msg)
+
+        gh_team_id = str(self._gh_interface.org_create_team(gh_team_name))
+        logging.debug(f"Github team {gh_team_name} created with "
+                      f"Github team ID {gh_team_id}")
+        team = Team(gh_team_id, gh_team_name, "")
+
+        if display_name is not None:
+            logging.debug(f"Attaching display name {display_name} "
+                          f"to {gh_team_name}")
+            team.display_name = display_name
+
+        if platform is not None:
+            logging.debug(f"Attaching platform {platform} to {gh_team_name}")
+            team.platform = platform
+
+        if channel is not None:
+            logging.debug(f"Adding channel members of #{channel} "
+                          f"to {gh_team_name}")
+            try:
+                channel_member_ids = \
+                    self._slack_client.get_channel_users(channel)
+                logging.debug(f"Member IDs of members found in #{channel}: "
+                              f"{channel_member_ids}")
+            except SlackAPIError as e:
+                msg = f"Channel member query on channel #{channel} failed: " \
+                    f"{e.error}"
+                logging.error(msg)
+                raise SlackAPIError(msg)
+            for member_id in channel_member_ids:
+                try:
+                    member = self._db_facade.retrieve(User, member_id)
+                    self._gh_interface.add_team_member(member.github_username,
+                                                       gh_team_id)
+                    team.add_member(member.github_id)
+                    logging.debug(f"Member with ID {member.slack_id} added "
+                                  f"to {gh_team_name}")
+                except LookupError:
+                    logging.warning(f"Member with ID {member_id} not found, "
+                                    "ignoring...")
+                    pass

```

Comment:
Probably a good idea to use a [bulk retrieve](https://rocket2.readthedocs.io/en/latest/docs/api/db.html#db.facade.DBFacade.bulk_retrieve) and check later if the lengths match.

Reply:
What would be bulk retrieved (and with what parameters)?

Reply:
```python
for member in self._db_facade.bulk_retrieve(channel_member_ids):
	pass
```

Reply:
Tried implementing what I thought you might want, let me know if there's anything to change in a follow up change request